smac/optimizer/ei_optimization.py

```diff
@@ -285,19 +271,31 @@ def _one_iter(
             all_neighbors = get_one_exchange_neighbourhood(
                 incumbent, seed=self.rng.randint(MAXINT))
 
+            neighbors = []
             for neighbor in all_neighbors:
                 s_time = time.time()
                 acq_val = self.acquisition_function([neighbor], *args)
                 neighbors_looked_at += 1
                 time_n.append(time.time() - s_time)
 
+                if acq_val == acq_val_incumbent:
+                    neighbors.append(neighbor)
                 if acq_val > acq_val_incumbent + self.epsilon:
                     self.logger.debug("Switch to one of the neighbors")
                     incumbent = neighbor
                     acq_val_incumbent = acq_val
                     changed_inc = True
                     break
 
+            if (
+                not changed_inc
+                and n_no_improvements < N_STEPS_PLATEAU_WALK
+                and len(neighbors) > 0
+            ):
+                n_no_improvements += 1
+                incumbent = neighbors[0]
```

Comment:
Can this be something else than all_neighbors[0]?

Reply:
Yes, only if the acquisition values are equal something is added to `neighbors`.